osu.Game/Screens/Select/BeatmapDetailArea.cs

```diff
@@ -32,7 +34,11 @@ public WorkingBeatmap Beatmap
             set
             {
                 beatmap = value;
-                if (IsLoaded) Schedule(updateScores);
+                if (IsLoaded)
+                    if (currentTab == BeatmapDetailTab.Details)
+                        Schedule(updateDetails);
+                    else
+                        Schedule(updateScores);

```

Comment:
I think this should always updateDetails. When you change to a scoring tab, change map then change to the details tab, it should already be in the updated state.